test/Microsoft.Health.Extensions.Fhir.R4.UnitTests/FhirClientValidatorTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Logging.Telemetry;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using Xunit;
+
+namespace Microsoft.Health.Extensions.Fhir.R4.UnitTests
+{
+    public class FhirClientValidatorTests
+    {
+        [Theory]
+        [InlineData("https://testfoobar.azurehealthcareapis.com")]
+        [InlineData("https://microsoft.com")]
+        public async Task GivenInvalidFhirServiceUrl_WhenValidateFhirService_ThenNotValidReturned_Test(string url)
+        {
+            await ValidateFhirClientUrl(url, false);
+        }
+
+        private async Task ValidateFhirClientUrl(string url, bool expectedIsValid)
+        {
+            var fhirClient = Substitute.For<IFhirClient>();
+
+            string uri = url + "/metadata";
+
+            fhirClient.ReadAsync<Hl7.Fhir.Model.CapabilityStatement>(uri).ThrowsForAnyArgs(new Exception());
```

Comment:
Thanks for taking the time to review Carrie. You are correct that a valid test case for the ValidateFhirClientUrl helper function would fail with these changes since the code is throwing an exception for any string parameter. We could update the mocked ReadAsync to return different results based on the test case urls. However, I've reverted my changes here so that we can add future tests cases without mocking out each Read call based on the URL

In regard to the naming, the ValidateFhirClientUrl helper function tests the Microsoft.Health.Fhir.R4.Client FhirClient rather than the FhirService class. We are similarly validating the Fhir Client in ValidateFhirClientAsync(). I think this naming helps to differentiate between FhirService.cs class and the FhirClient 

Happy to discuss further